compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1878,9 +1878,10 @@ public BLangNode transform(ListConstructorExpressionNode listConstructorExprNode
     public BLangNode transform(UnaryExpressionNode unaryExprNode) {
         Location pos = getPosition(unaryExprNode);
         SyntaxKind expressionKind = unaryExprNode.expression().kind();
-        if (expressionKind == SyntaxKind.NUMERIC_LITERAL) {
-            BLangNumericLiteral numericLiteral = (BLangNumericLiteral) createSimpleLiteral(unaryExprNode);
-            return numericLiteral;
+        SyntaxKind unaryOperatorKind = unaryExprNode.unaryOperator().kind();
+        if (expressionKind == SyntaxKind.NUMERIC_LITERAL &&
+                         (unaryOperatorKind == SyntaxKind.MINUS_TOKEN || unaryOperatorKind == SyntaxKind.PLUS_TOKEN)) {
+            return createSimpleLiteral(unaryExprNode);
```

Comment:
Do we need to do this at all?

+(1) and -(1) would not satisfy the if condition, so the path would be different for them vs +1 and -1?

Can you please create an issue for this?

Reply:
Created(https://github.com/ballerina-platform/ballerina-lang/issues/30020)